tests/test.rs

```diff
@@ -155,690 +155,6 @@ fn test_clone_reader() {
     assert!(matches!(cloned.read_event().unwrap(), End(_)));
 }
 
-#[cfg(feature = "serialize")]
```

Comment:
The `seq` and `struct` tests feel rather abstract to me, because `seq` tests only look at the number of elements without checking the contents, and `struct` tests only check the serialization of one element at a time.

This is more of an integration test which covers a much more complex scenario involving multiple elements on the same struct and nesting, which the other tests don't appear to cover thoroughly.  

I'd personally be more comfortable if it was moved to `serde_roundtrip` but if there's a strong case that it doesn't add value I would listen

Reply:
Yes, this is possible to add a special test for `seq + non-trivial struct` to existing `seq` tests to test an integration. I don't want to restore this specific test because it is too big and redundant to test such a scenario. If it falls, it will be difficult to find an error.

Reply:
Added new tests